src/commonMethods/webKitPerformanceCommonFunctions.js

```diff
@@ -0,0 +1,61 @@
+let samples = []
+
+/**
+ * This function resets the Samples array
+ */
+export const initializeSamplesArray = function() {
+  samples = []
+}
+
+/**
+ * This function fetches the FPS value
+ * @returns {Samples Array}
+ */
+export const fetchFPS = function() {
+  this.$thunder.api.WebKitBrowser.fps()
+    .then(result => {
+      if (isNaN(result) === false) {
+        samples.push(result)
+        return true
+      } else {
+        this.$log('Returned FPS value from Framework is not a number')
+        throw new Error('Returned FPS value from Framework is not a number')
+      }
+    })
+    .catch(err => err)
+  return samples
+}
+
+/**
+ * This function calculates the average of FPS samples collected in fetchFPS function
+ * @returns {results}
+ */
+export const calcAvgFPS = function() {
+  let sum = 0
+  let results = {}
+  for (let i = 0; i < samples.length; i++) {
+    sum += samples[i]
+  }
+  results.average = sum / samples.length
```

Comment:
why an object with an `average`-key instead of just returning the value?

Reply:
We have written this test as per the old test case. Now we have updated this by removing results object and directly taking the variable 'average '